File: Integraupt-Backend/BackendAdReserva/src/main/java/com/AdReserva/dto/AdminGestionReservaRequest.java
package com.AdReserva.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record AdminGestionReservaRequest(
        @NotNull(message = "El identificador del usuario gestor es obligatorio")
        Integer usuarioGestionId,
        @NotBlank(message = "La acción a ejecutar es obligatoria")
        String accion,
        @NotBlank(message = "Debes registrar un motivo para la gestión realizada")
        @Size(max = 1000, message = "El motivo no debe superar los 1000 caracteres")
        String motivo,
        @Size(max = 1000, message = "Los comentarios no deben superar los 1000 caracteres")
        String comentarios
) {
}

File: Integraupt-Backend/servicio-adreserva/src/DTO/AdminGestionReservaRequest.php
<?php

namespace App\DTO;

class AdminGestionReservaRequest {
    public ?int $usuarioGestionId;
    public ?string $accion;
    public ?string $motivo;
    public ?string $comentarios;

    public function __construct(?int $usuarioGestionId, ?string $accion, ?string $motivo, ?string $comentarios) {
        $this->usuarioGestionId = $usuarioGestionId;
        $this->accion = $accion;
        $this->motivo = $motivo;
        $this->comentarios = $comentarios;
    }
}
